pkgs/unified_analytics/lib/src/session.dart

```
@@ -9,13 +9,17 @@ import 'package:file/file.dart';
 import 'package:path/path.dart' as p;
 
 import 'constants.dart';
+import 'initializer.dart';
 
 class Session {
   final Directory homeDirectory;
   final FileSystem fs;
   final File _sessionFile;
-  int _sessionId;
-  int _lastPing;
+
+  // Initialize as 0 but will get parsed to real values
```

@christopherfujino i wanted to avoid using the late keyword here for these 2 variables, let me know what you think

---

If we're calculating this within the private constructor, can we not calculate it in the factory?

And if not, can we go the other way and get rid of the factory altogether?

---

There is an instance method `_refreshSessionData()` that will be used to calculate those two ints so it can't be used within the factory constructor body.

But I agree with removing the factory constructor

---

Out of curiosity, why using avoid `late` here?

---

I was referring to the usage guide where it says to avoid `late` where possible. I also have used it in other parts of the package that could probably be refactored once this package is stable and being used

https://dart.dev/guides/language/effective-dart/usage#avoid-late-variables-if-you-need-to-check-whether-they-are-initialized

---

From the link you shared:

> AVOID late variables _**if you need to check whether they are initialized**_

I think using `late` is fine if a variable should definitely be initialized before it's read. While I believe avoiding `late` by default is good, I think initializing variables with placeholder values has more (potential) problems. If we intend for a variable to get "properly" initialized before first read, then reading it before initialization should throw, and `late` does this for us.

---

That's a great point, I also agree with that. I'll revert that back to use `late` again

---

LGTM